src/commands/tag.js

```diff
@@ -0,0 +1,114 @@
+import { FileSystem } from '../models/FileSystem.js'
+import { join } from '../utils/join.js'
+import { cores } from '../utils/plugins.js'
+import { GitAnnotatedTag } from '../models/GitAnnotatedTag'
+
+/**
+ * Create a lightweight or annotated tag.
+ *
+ * @link https://isomorphic-git.github.io/docs/tag.html
+ */
+export async function tag ({
+  core = 'default',
+  dir,
+  gitdir = join(dir, '.git'),
+  fs: _fs = cores.get(core).get('fs'),
+  name,
+  annotated = undefined,
+  value = undefined,
+  force = false
+}) {
+  try {
+    const fs = new FileSystem(_fs)
+    const ref = 'refs/tags/' + name
+
+    // Resolve passed value
+    value = await GitRefManager.resolve({
+      fs,
+      gitdir,
+      ref: value || 'HEAD'
+    })
+
+    try {
+      // Attempt to get old value
+      const oldValue = await GitRefManager.resolve({ fs, gitdir, ref })
+      // If found and not forced overwriting
+      if (!force) {
+        if (oldValue !== value) {
+          // It exists and differ from passed value
+          throw new GitError(E.RefExistsError, { noun: 'tag', ref: name })
+        } else {
+          // Nothing to do, as the old value and the new one is the same
+          return
+        }
+      }
+    } catch (err) {
+      switch (err.name) {
+        case E.ResolveRefError:
+          // Not found, it's pretty OK
+          break
+        default: throw err
+      }
+    }
+
+    if (annotated) {
+      const message = annotated.message
+      let tagger = annotated.tagger
+      const signingKey = annotated.signingKey
+      // Fill in missing arguments with default values
+      if (tagger === undefined) tagger = {}
+      if (tagger.name === undefined) {
+        tagger.name = await config({ fs, gitdir, path: 'user.name' })
+      }
+      if (tagger.email === undefined) {
+        tagger.email = await config({ fs, gitdir, path: 'user.email' })
+      }
+      if (tagger.name === undefined || tagger.email === undefined) {
+        throw new GitError(E.MissingAuthorError)
+      }
+      if (message === undefined) {
+        throw new GitError(E.MissingRequiredParameterError, {
+          function: 'tag',
+          parameter: 'message'
+        })
+      }
+      const referredObjectType = (await readObject({ fs, gitdir, oid: value })).type
+      let taggerDateTime = tagger.date || new Date()
+      let tag = GitAnnotatedTag.from({
+        object: value,
+        type: referredObjectType,
+        tag: name,
+        tagger: {
+          name: tagger.name,
+          email: tagger.email,
+          timestamp:
+              tagger.timestamp !== undefined && tagger.timestamp !== null
+                ? tagger.timestamp
+                : Math.floor(taggerDateTime.valueOf() / 1000),
+          timezoneOffset:
+              tagger.timezoneOffset !== undefined &&
+              tagger.timezoneOffset !== null
+                ? tagger.timezoneOffset
+                : new Date().getTimezoneOffset()
```

Comment:
this should probably be taggerDateTime.getTimezoneOffset(). Looks like the original bug is in `commit.js` so that's on me.